src/coreclr/debug/createdump/crashinfo.cpp

```diff
@@ -6,10 +6,13 @@
 // This is for the PAL_VirtualUnwindOutOfProc read memory adapter.
 CrashInfo* g_crashInfo;
 
-CrashInfo::CrashInfo(pid_t pid) :
+CrashInfo::CrashInfo(pid_t pid, bool crashReport, pid_t crashThread, uint32_t signal) :
     m_ref(1),
     m_pid(pid),
-    m_ppid(-1)
+    m_ppid(-1),
+    m_crashReport(crashReport),
```

Comment:
I passed to the crash info because there are expensive things like the stack frames that are gathered only for crash reporting.  Maybe the dumpwriter (or the crashreportwriter) should also be passed this flag instead of using the public property on crashinfo.

Reply:
Yes. CrashInfo should probably get something like `collectAdditionalDiagnostics` if `crashReport` is true

Reply:
I don't think it is worth it to separate/rename this flag.

Reply:
I changed my mine. I changed it to "gatherFrames" since that is only CrashInfo does if true.